casadm/cas_lib.c

```diff
@@ -547,6 +550,29 @@ void print_slow_atomic_cache_start_info(const char *device_path)
 	}
 }
 
+/**
+  * Save to dest an absolute device file path of src.
+  * Return number of characters copied to dest if succeed, negative value if failed.
+  */
+static int get_abs_path(char* dest, const char* src,  size_t max_len)
+{
+	int path_len;
+	char *dir_name, *dev_name;
+	char *dev = strdup(src);  // strdup creates hidden malloc
+	char *dir = strdup(src);  // basename/dirname may modify the source string
+
+	dir_name = realpath(dirname(dir), NULL); // realpath creates hidden malloc
+	dev_name = basename(dev);
+
+	path_len = snprintf(dest, max_len, "%s/%s", dir_name, dev_name);
+
+	free(dev);
+	free(dir);
+	free(dir_name);
+
+	return path_len;
+}
+
```

Comment:
isn't this all equivalent to realpath?

Reply:
Nope, `realpath` would resolve path, if given path is link. This won't resolve it, but return full absolute path without redundant characters like `/` or `.`.

Reply:
Right, I remember now - thanks :)